sql3/sql_complex_test.go

```diff
@@ -3038,3 +3039,77 @@ func TestPlanner_BulkInsert_FP1916(t *testing.T) {
 	expected := int64(8924809397503602651)
 	assert.Equal(t, got, expected)
 }
+
+func TestPlanner_BulkInsert_FP1915(t *testing.T) {
+	c := test.MustRunCluster(t, 3)
+	defer c.Close()
+	// 8924809397503602651 is larger than 2^53 which is the largest integer value representable in float64
+	node := c.GetNode(0).Server
+	_, _, _, err := sql_test.MustQueryRows(t, node, `create table greg-test (
+		_id STRING,
+		id_col ID,
+		string_col STRING cachetype ranked size 1000,
+		int_col int,
+		decimal_col DECIMAL(2),
+		bool_col BOOL
+		time_col TIMESTAMP,
+		stringset_col STRINGSET,
+		ideset_col IDSET
+	);`)
+	assert.NoError(t, err)
+
+	_, _, _, err = sql_test.MustQueryRows(t, node, `BULK INSERT INTO greg-test (
+		_id,
+		id_col,
+		string_col,
+		int_col,
+		decimal_col,
+		bool_col,
+		time_col,
+		stringset_col,
+		ideset_col)
+		map (
+		'id_col' ID,
+		'string_col' STRING,
+		'int_col' INT,
+		'decimal_col' DECIMAL(2),
+		'bool_col' BOOL,
+		'time_col' TIMESTAMP,
+		'stringset_col' STRINGSET,
+		'ideset_col' IDSET)
+		transform(
+		CAST(@0 as STRING),
+		@0,
+		@1,
+		@2,
+		@3,
+		@4,
+		@5,
+		@6,
+		@7)
+		FROM x'{"id_col": "8924809397503602651", "string_col": "TEST", "decimal_col": "1.12", "bool_col": false, "time_col": "2013-07-15T01:18:46Z", "stringset_col": "stringset1","ideset_col": 1}
+		{"id_col": "8924809397503602652", "string_col": "TEST2", "decimal_col": "1.12", "bool_col": false, "time_col": "2013-07-15T01:18:46Z", "stringset_col": ["stringset1","stringset3"],"ideset_col": [1,2]}
+		{"id_col": "8924809397503602653", "string_col": "TEST", "int_col": "321", "decimal_col": "12.1", "bool_col": 1, "time_col": "2014-07-15T01:18:46Z", "stringset_col": "stringset2","ideset_col": [1,3]}
+		'
+		with
+			BATCHSIZE 10000
+			format 'NDJSON'
+			input 'STREAM';
+	`)
+	assert.NoError(t, err)
+	results, _, _, err := sql_test.MustQueryRows(t, node, `select _id from greg-test`)
+	assert.NoError(t, err)
+	vprint.VV("results %#v", results)
+	if diff := cmp.Diff([][]interface{}{
+		{"8924809397503602651"},
+		{"8924809397503602652"},
+		{"8924809397503602653"},
+	}, results); diff != "" {
+		t.Fatal(diff)
+	}
+	/*
```

Comment:
i think we can drop this commented-out version, but apart from that this looks reasonable?

although i'm a bit confused because `id_col` is the input value, and then we're doing `select _id`, but i guess that's the ID? this may actually be the largest shard number we've ever tried to use.